packages/react-router/lib/hooks.tsx

```diff
@@ -939,7 +939,6 @@ export function useBlocker(shouldBlock: boolean | BlockerFunction): Blocker {
   let state = useDataRouterState(DataRouterStateHook.UseBlocker);
 
   let [blockerKey, setBlockerKey] = React.useState("");
-  let [blocker, setBlocker] = React.useState<Blocker>(IDLE_BLOCKER);
```

Comment:
Per https://github.com/remix-run/remix/issues/6704#issuecomment-1614413959, it was pointed out that we don't even need the blocker in local state anymore.  We prefer the `state.blockers` value anyway as the source of truth that updates via `DataRouterStateContext` and we can just fallback on `IDLE_BLOCKER` instead of syncing router state into local state.